src/diffusers/utils/hub_utils.py

```diff
@@ -359,3 +370,210 @@ def _get_model_file(
                 f"Otherwise, make sure '{pretrained_model_name_or_path}' is the correct path to a directory "
                 f"containing a file named {weights_name}"
             )
+
+
+# Taken from
+# https://github.com/huggingface/transformers/blob/5bb4430edc7df9f9950d412d98bbe505cc4d328b/src/transformers/utils/generic.py#L453
+@contextmanager
```

Comment:
Don't think a context manager is a good design here. Context managers should be used mostly for closing things not for moving working directories. Can we just use an if-else instead?